lib/liquid/context.rb

```diff
@@ -47,6 +48,22 @@ def add_filters(filters)
       @strainer = nil
     end
 
+    def add_global_filters(global_filters)
+      gf = [global_filters].flatten.compact
```

Comment:
There is no need to have multiple procs get called.  The proc that gets passed in can just call multiple methods.